### PR TITLE
feat(viz): added seg map viz tool

### PR DESCRIPTION
**Pull request by kartik4949**
close #318 

Note:
Conversation history: #220

### CODE REVIEW

**Comment by LukeWood** [comment]
HI @kartik4949 - I made some edits already in the other PR.  Can you pull those back in?

**Comment by kartik4949** [comment]
@LukeWood  I cherry picked them.

**Comment by kartik4949** [comment]
@LukeWood Done, some changes wrt linting and tests, you can take the pr from now on, Thanks a lot for help throughout the PR
Cheers!

**Comment by kartik4949** [comment]
Updated the branch on latest master commit

File: keras_cv/visualization/mask.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+        Masks overlaid on images.
+
+    Raises:
+        ValueError: On incorrect data type and shapes for images or masks.
+        KeyError: On incorrect color string in `color`, or a class ID missing from
+        the color map.
+        TypeError: On incorrect color type.
+
+    References:
+    - [KerasCV Colors](https://tinyurl.com/34bm28zu)
```
Comment (kartik4949) [inline keras_cv/visualization/mask.py:120]: hmm.. sweet!

**Comment by LukeWood** [comment]
Hey @kartik4949; as time goes on it becomes clearer and clearer to me that we need to (as a library) be *very* careful about the formats we pick to represent complex structures; including bounding boxes and segmentation maps.

As such, I'm hesitant to merge a segmentation map feature until I have done *significant* research into segmentation map modeling and representation.

I'm happy to merge the `colors.py` file, but I'd like to hold off on this PR until we do this.  Feel free to leave this open, and feel free to open a PR adding `visualization/colors.py` and `visualization/__init__.py` importing `colors.py`. 

Thanks @kartik4949 !

**Comment by kartik4949** [comment]
@LukeWood Make sense!, sure ill try to pr colors.py

**Comment by bhack** [comment]
We need to handle format and its utils with care.  
In some case we need to handle also transparency like in matting, or  splines and polygons, in other case we need to extend this format with temporal/id information etc..

https://www.v7labs.com/open-datasets/composition-1k
https://arxiv.org/abs/2204.09433
https://github.com/fidler-lab/curve-gcn
https://motchallenge.net/data/MOTSynth-MOTS-CVPR22/

**Comment by LukeWood** [comment]
> We need to handle format and its utils with care. In some case we need to handle also transparency like in matting, or splines and polygons, in other case we need to extend this format with temporal/id information etc..
> 
> https://www.v7labs.com/open-datasets/composition-1k https://arxiv.org/abs/2204.09433 https://github.com/fidler-lab/curve-gcn https://motchallenge.net/data/MOTSynth-MOTS-CVPR22/

Agreed.  I don't want to offer any segmentation map APIs until we can figure out the format to use.

**Comment by LukeWood** [comment]
Hey @kartik4949 !  I really appreciate your contributions to KerasCV.  We have decided to NOT branch out into visualization tools just yet.  If we do, I'd like to contribute the first component myself so I can think carefully about the API and make those decisions in a vacuum.  Sorry about that, and I hope you contribute again in the future.

**Pull request closed without merging**